### ISSUE TITLE
[vmauth] http: superfluous response.WriteHeader

### ISSUE THREAD

**Issue by zbindenren** [VictoriaMetrics/VictoriaMetrics#1078]
**Describe the bug**
Updating vmauth to 1.54.1 breaks all our grafana dashboards with:

```
Failed to load resource: net::ERR_CONTENT_DECODING_FAILED
```

In the vmauth logs I see the following:

```
Feb 19 13:19:50 t1-vmselect-alsu101 vmauth[3889450]: 2021-02-19T12:19:50.836Z        error        net/http/server.go:3137        http: superfluous response.WriteHeader call from github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.(*gzipResponseWriter).writeHeader (httpserver.go:423)
Feb 19 13:19:50 t1-vmselect-alsu101 vmauth[3889450]: 2021-02-19T12:19:50.839Z        error        net/http/server.go:3137        http: superfluous response.WriteHeader call from github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.(*gzipResponseWriter).writeHeader (httpserver.go:423)
Feb 19 13:19:50 t1-vmselect-alsu101 vmauth[3889450]: 2021-02-19T12:19:50.842Z        error        net/http/server.go:3137        http: superfluous response.WriteHeader call from github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.(*gzipResponseWriter).writeHeader (httpserver.go:423)
Feb 19 13:19:50 t1-vmselect-alsu101 vmauth[3889450]: 2021-02-19T12:19:50.842Z        error        net/http/server.go:3137        http: superfluous response.WriteHeader call from github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.(*gzipResponseWriter).writeHeader (httpserver.go:423)
Feb 19 13:19:50 t1-vmselect-alsu101 vmauth[3889450]: 2021-02-19T12:19:50.860Z        error        net/http/server.go:3137        http: superfluous response.WriteHeader call from github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.(*gzipResponseWriter).writeHeader (httpserver.go:423)
Feb 19 13:19:50 t1-vmselect-alsu101 vmauth[3889450]: 2021-02-19T12:19:50.860Z        error        net/http/server.go:3137        http: superfluous response.WriteHeader call from github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.(*gzipResponseWriter).writeHeader (httpserver.go:423)
Feb 19 13:19:50 t1-vmselect-alsu101 vmauth[3889450]: 2021-02-19T12:19:50.863Z        error        net/http/server.go:3137        http: superfluous response.WriteHeader call from github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.(*gzipResponseWriter).writeHeader (httpserver.go:423)
Feb 19 13:19:50 t1-vmselect-alsu101 vmauth[3889450]: 2021-02-19T12:19:50.869Z        error        net/http/server.go:3137        http: superfluous response.WriteHeader call from github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.(*gzipResponseWriter).writeHeader (httpserver.go:423)
Feb 19 13:21:37 t1-vmselect-alsu101 vmauth[3889450]: 2021-02-19T12:21:37.108Z        error        net/http/server.go:3137        http: superfluous response.WriteHeader call from github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.(*gzipResponseWriter).writeHeader (httpserver.go:423)
```

Downgrading to 1.53.x fixes the issue again.

**To Reproduce**
Update vmauth to 1.154.1. Open a Grafana dashboard with the corresponding data source. Dashboard loading fails.

**Expected behavior**
Dashboard should work.

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```console
$ vmauth --version
vmauth-20210218-171608-tags-v1.54.1-0-g5e341ccb5
```

**Used command-line flags**
```
httpListenAddr=:7001
auth_config=/appl/vmauth/config.yaml
tls=true
tlsCertFile=/path/to/ertificate.pem
tlsKeyFile=/path/to/key_nocrypt.pem
http_shutdownDelay=45s
```

```yaml
users:
- username: 'user'
  password: '***'
  url_prefix: 'http://localhost:8481/select/0/prometheus'
```

**Comment by transacid** [comment]
appending `-http.disableResponseCompression` to the startup arguments helped me for the time being. judging bey the error message it seems like there is a problem with the gzip compression

**Comment by zbindenren** [comment]
@transacid thanks, this resolves my issue too.

**Comment by valyala** [comment]
The issue must be fixed in the commit 9fc7726d848893d4542f44ef4c1becc208fd5666 . @zbindenren , @transacid , could you build `vmauth` from this commit according to [these docs](https://victoriametrics.github.io/vmauth.html#how-to-build-from-sources) and verify the issue is gone?

**Comment by transacid** [comment]
can confirm,  9fc7726 fixes the issue for me. thanks!

**Comment by valyala** [comment]
@transacid , thanks for the confirmation!

The bugfix is available starting from [v1.55.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
